firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -880,21 +909,24 @@ def pbs_for_set(document_path, document_data, option):
         or two ``Write`` protobuf instances for ``set()``.
     """
     transform_paths, actual_data = remove_server_timestamp(document_data)
-
     update_pb = write_pb2.Write(
         update=document_pb2.Document(
             name=document_path,
             fields=encode_dict(actual_data),
         ),
     )
     if option is not None:
-        option.modify_write(update_pb)
+        field_paths = extract_field_paths(actual_data)
+        option.modify_write(update_pb, field_paths=field_paths)
```

Comment:
I still think we should only do this when merge is set.

Reply:
I'll take a look, but field_paths is ignored for the ExistsOption and LastUpdateOptions.

Reply:
I have looked and tried to do this for `isinstance(option, MergeOption)` and resulted in no change.  I think I will just duck type this instead of checking for `isinstance`.  If other options will change in the future, we can deal with it when we need it easily.

Reply:
Maybe you have some situation that I'm missing?

Reply:
Looking at the api file, maybe I should implement the options differently.  But I think the behavior for this case would be correct.

I'll work on the options in a bit.